packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/GeoBox.kt

```diff
@@ -0,0 +1,37 @@
+package io.realm.kotlin.types
+
+/**
+ * This class represents a rectangle on the surface of a sphere. It cannot be persisted - i.e you
+ * can't declare a Realm property that is of type [GeoBox]. It can only be used as a query argument
+ * for a geospatial query like this:
+ *
+ * ```
+ * val bottomleft = GeoPoint(latitude = 0.0, longitude = 0.0)
+ * val topRight = GeoPoint(latitude = 0.0, longitude = 0.0)
+ * val searchArea = GeoBox(bottomLeft, topRight)
+ * val restaurants = realm.query<Restaurant>("location GEOWITHIN $searchArea").find()
+ * ```
+ */
+public data class GeoBox(
+    /**
+     * The bottom left corner of the rectangle.
+     */
+    public val bottomLeft: GeoPoint,
+    /**
+     * The top right corner of the rectangle.
+     */
+    public val topRight: GeoPoint
+) {
+    /**
```

Comment:
I guess we can also do that.

Reply:
Arg, my bad, I meant double. I was thinking of having a constructor that doesn't require to instantiate GeoPoints
```
GeoBox(
    latitude1: Double,
    longitude1: Double,
    latitude2: Double,
    longitude2: Double
)
```

Reply:
Ah, I would really like to avoid that, it looks way too easy to mess up the order of arguments IMO. I agree that the current constructor might look a bit verbose, but it is very easy to see what is happening.